### PR TITLE
Fix #49, @showprogress macro under julia 0.5.

### LINKED ISSUE #49: Tests fail on nightly
0.5-style comprehensions changed `loopassign.head` to `:generator` in some cases, see http://pkg.julialang.org/detail/ProgressMeter.html
```
Testing @showprogress macro on for loop
...
[1GProgress:  37%  ETA: 0:00:09[K[1GProgress:  40%  ETA: 0:00:08[K[1GProgress:  43%  ETA: 0:00:07[K[1GProgress:  47%  ETA: 0:00:06[K[1GProgress: 100% Time: 0:00:05[K
ERROR: LoadError: LoadError: AssertionError: loopassign.head === :(=)
 in include_from_node1(::String) at ./loading.jl:426 (repeats 2 times)
 in process_options(::Base.JLOptions) at ./client.jl:266
 in _start() at ./client.jl:322
while loading /home/vagrant/.julia/v0.5/ProgressMeter/test/test.jl, in expression starting on line 106
```

Just changing the assertion to be more permissive (`VERSION` for the 0.5-style comprehensions merge commit is `v"0.5.0-dev+5297"`) doesn't seem to fix matters, so it might be a bit complicated to fix.

### PR DESCRIPTION
**Pull request by garrison**

### CODE REVIEW

**Comment by garrison** [comment]
I didn't make an effort to determine the precise `VERSION` at which this change was made, but I don't think it really matters; the important thing is that `@showprogress` now works on the 0.5 release candidates.

There are still four warnings under 0.5, however, [given here](https://travis-ci.org/timholy/ProgressMeter.jl/jobs/148496309#L193-L203):

    WARNING: deprecated syntax "(Int=>Int)[a=>b for (a,b) in c]" at /home/travis/.julia/v0.5/ProgressMeter/test/test.jl:193.
    Use "Dict{Int,Int}(a=>b for (a,b) in c)" instead.
    WARNING: deprecated syntax "(Int=>Int)[a=>b for (a,b) in c]" at /home/travis/.julia/v0.5/ProgressMeter/test/test.jl:194.
    Use "Dict{Int,Int}(a=>b for (a,b) in c)" instead.
    WARNING: deprecated syntax "(Tuple{Int,Int}=>Int)[a=>b for (a,b) in c]" at /home/travis/.julia/v0.5/ProgressMeter/test/test.jl:199.
    Use "Dict{Tuple{Int,Int},Int}(a=>b for (a,b) in c)" instead.
    WARNING: deprecated syntax "(Tuple{Int,Int}=>Int)[a=>b for (a,b) in c]" at /home/travis/.julia/v0.5/ProgressMeter/test/test.jl:200.
    Use "Dict{Tuple{Int,Int},Int}(a=>b for (a,b) in c)" instead.

Unfortunately if I make these suggested changes, I get syntax errors on julia 0.4, even if I wrap with the `@compat` macro...

    ERROR: LoadError: LoadError: syntax: missing comma or ) in argument list

**Comment by timholy** [comment]
Because it's a parser change there's no way to use `@compat`. You can construct with an array of 2-tuples rather than array of `Pair`s, I believe.

**Comment by timholy** [comment]
LGTM. Merge whenever you're ready.

File: src/ProgressMeter.jl [hunk 1]
```diff
@@ -420,7 +421,14 @@ macro showprogress(args...)
         throw(ArgumentError("Final argument to @showprogress must be a for loop or comprehension."))
     end
 
-    loop = copy(loop)
+    # In julia 0.5, a comprehension's "loop" is actually one level deeper in the syntax tree.
+    if VERSION >= v"0.5.0-rc0" && loop.head !== :for
```
Comment (tkelman) [inline src/ProgressMeter.jl]: probably `v"0.5.0-dev+5297"` as I wrote down in the issue?

Comment (garrison) [inline src/ProgressMeter.jl]: Thanks, I had forgotten you had tracked that down.

**Pull request merged**